### DISCUSSION TITLE
ComicTagger plugin

### DISCUSSION THREAD

**Discussion by mizaki** [Metron-Project/metroninfo#43] [General]
It seems this has ramped up and is close to release? 

Has anyone started a ComicTagger plugin for the format yet? Copying https://github.com/comictagger/comicinfoxml should be easy enough. If not I can add it to my list of too many things to do and not enough time (as I'm sure you have the same).

**Comment by bpepple** [comment]
With any luck we should be finalizing the v1 release fairly soon (maybe this weekend or early next week), I just need to touch base with Joe (Kavita) and AJ (Codex) to make sure they don't have any outstanding issues that need to be resolved.

I've almost finished adding support in our tools (darkseid, metron-tagger) in the development branches, tho I'm thinking of writing a separate tool for users to handle adding the `MetronInfo.xml` to the their existing collection without absolutely hammering our server. Most likely it will be simple command line tool that does thing like using any information in their existing `ComicInfo.xml` to get the comic vine or metron id and use that to query the server since it greatly reduce the number of API calls, and also have the rate limit to be adjustable to a number less than our maximum that way they could just have it running in the background using `screen` or any other terminal multiplexer.

I haven't looked at CT in awhile, but I'm guessing your metadata class would need to be modified to handle the additional information that `ComicInfo.xml` doesn't capture (alternative comic id's, resource id's, prices, etc). And truthfully, if CT adds support at later time that's not a bad thing since it would reduce the load on our server by spreading out people updating their collections. If someone doesn't get to it first I'm might look at doing it, tho it would most likely be in late October.

**Comment by mizaki** [comment]
I've not looked into the metadata plugin side much yet but internally https://github.com/comictagger/comictagger/blob/develop/comicapi/genericmetadata.py most are handled. The IDs on the resources as is would not be added only kept if already present I would imagine. I also don't see an issue with multiple source IDs but it would have to be hardcoded currently if a `primary` is already present if it remains or the new source subsumes.

ComicTagger does have a Copy Tags function fyi. :)

Whatever you want to do. If you want to create it yourself to keep it in the Metron garden happy to help if needed. If you want to have it on my list, that's fine, and I'm happy to give you a poke before setting it free if so.

**Reply by lordwelch** [reply]
I don't see CT ever supporting the resource ID's as trying to resolve that when multiple sources are involved is way too much work. There is at least one person I have seen that does a run on GCD and then CV (when it's working) and that would be a nightmare to try and merge those IDs. Preserving them would work for single elements (Publisher) as is if the code for ComicInfo.xml is used as a base, but for lists you would need to search the list for each value and then update that field and so I'm not sure it would really work out well. I'm also not sure about adding new attributes in CT for Universe and Reprint

**Comment by bpepple** [comment]
@lordwelch & @mizaki, we are looking for input on whether to add the series status and issue count to the schema, and if you have an opinion on it we'd love to get your input.

https://github.com/Metron-Project/metroninfo/issues/38

**Comment by mizaki** [comment]
I know you wanted to hold off on CT support, is that still the case?

**Reply by bpepple** [reply]
No,  the spec is finalized, and some apps have already added support for it,  and others have it on their to-do lists, so it's fine for CT to add support.
